pints/_optimisers/_adam.py

```diff
@@ -12,16 +12,36 @@
 
 class Adam(pints.Optimiser):
     """
-    Adam optimiser (adaptive moment estimation), as described in [1]_.
+    Adam optimiser (adaptive moment estimation), as described in [1]_ (see
+    Algorithm 1).
 
     This method is a variation on gradient descent that maintains two
     "moments", allowing it to overshoot and go against the gradient for a short
-    time. This property can make it more robust against noisy gradients. Full
-    pseudo-code is given in [1]_ (Algorithm 1).
+    time. This property can make it more robust against noisy gradients.
 
-    This implementation uses a fixed step size, set as `` min(sigma0)``. Note
-    that the adaptivity in this method comes from the changing moments, not
-    the step size.
+    Pseudo-code is given below. Here the value of the j-th parameter at
+    iteration i is given as ``p_j[i]`` and the corresponding derivative is
+    denoted ``g_j[i]``::
+
+        b1[i] = b1[i - 1] * beta1
+        b2[i] = b2[i - 1] * beta2
```

Comment:
I am a little bit unsure how b1 and b2 are initialised from this documentation. Is b1[i] = beta1**i, so b1[0]=1? Or is something more fancy happening?

Reply:
Ok yes, from looking at the code, I can see that is what is happening. 

Could we maybe simplify the documentation then and just write b1[i] = beta^i, which avoids defining initial conditions separately?

Reply:
Thanks @DavAug . Yes the pseudo code was a bit confusing and wrong (or maybe better to say that I was confused and wrong). Have updated it now (but with beta^(1+i) to account for my i=0 equalling the mathsy t=1)